### PR TITLE
Tooltip accessibility improvements: focus and voice over

### PR DESCRIPTION
**Pull request by alekzvik**
This PR addresses #8111 & #8112.

I've used the approach described in mentioned issues. I had to add logic for checking if the layer we're working with is a single `Layer` (one Marker, Polygon, etc.) or a `LayerGroup` (e.g., a group of Markers).

Changes include:
1. The layer a tooltip is bound to now has an `aria-describedby` attribute
2. Every tooltip has an id to accommodate the change above
3. Tooltips are open when acquiring focus and close on blur

Here is a before/after demo on a `debug/map/tooltip.html` with Voice Over narration visible:
<details><summary>Before</summary>

</details>

<details><summary>After</summary>

</details>

### CODE REVIEW

**Comment by Malvoz** [comment]
Will need others to review the logic, but this looks really good IMO, @alekzvik!

File: src/layer/Tooltip.js [hunk 1]
```diff
@@ -115,6 +117,9 @@ export var Tooltip = DivOverlay.extend({
 		    className = prefix + ' ' + (this.options.className || '') + ' leaflet-zoom-' + (this._zoomAnimated ? 'animated' : 'hide');
 
 		this._contentNode = this._container = DomUtil.create('div', className);
+
+		this._container.role = 'tooltip';
+		this._container.setAttribute('id', 'tooltip-' + Util.stamp(this));
```
Comment (Malvoz) [inline src/layer/Tooltip.js:122]: I believe we'll want a `leaflet-tooltip-` prefix on these IDs.

File: src/layer/Tooltip.js [hunk 2]
```diff
@@ -115,6 +117,9 @@ export var Tooltip = DivOverlay.extend({
 		    className = prefix + ' ' + (this.options.className || '') + ' leaflet-zoom-' + (this._zoomAnimated ? 'animated' : 'hide');
 
 		this._contentNode = this._container = DomUtil.create('div', className);
+
+		this._container.role = 'tooltip';
```
Comment (Malvoz) [inline src/layer/Tooltip.js:121]:
```suggestion
		this._container.setAttribute('role', 'tooltip');
```

File: src/layer/Tooltip.js [hunk 3]
```diff
@@ -341,6 +357,27 @@ Layer.include({
...
+	_addFocusListeners: function () {
+		if (this.getElement) {
+			this._addFocusListenersOnLayer(this);
+		} else if (this.eachLayer) {
+			this.eachLayer(this._addFocusListenersOnLayer, this);
+		}
+	},
+
+	_addFocusListenersOnLayer: function (layer) {
+		DomEvent.on(layer.getElement(), 'focus', () => {
```
Comment (Malvoz) [inline src/layer/Tooltip.js:369]: Arrow functions are not supported in IE.

Comment (Malvoz) [inline src/layer/Tooltip.js:369]: Arrow functions are not supported in IE.

**Review by Malvoz — APPROVED** [review]

**Review by mourner — APPROVED** [review]
👍 Looks great to me!

**Comment by MartijnHarmenzon** [comment]
Hi all. Is there a way to prevent this behaviour?

I have CircleMarkers of which the color changes when clicked on. But now they are getting the black and white outlines?

**Comment by jonkoops** [comment]
@MartijnHarmenzon if this change is causing problems for you could you create a new issue with a reproducible case? This makes things easier to keep track of for us.

**Comment by Malvoz** [comment]
> I have CircleMarkers of which the color changes when clicked on (using the setStyle function). But now they are getting the black and white outlines?

@MartijnHarmenzon The outlines (or other [obvious focus styles](https://developer.mozilla.org/en-US/docs/Web/CSS/outline#accessibility_concerns)) are essential for keyboard-only users when navigating features. You can keep the outlines only if the UA determines them "necessary" (pretty much boils down to programmatic focus + keyboard focus) by utilizing [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) like so:

```css
.leaflet-container :focus:not(:focus-visible) {
  outline: 0;
}
```

**Comment by FloChehab** [comment]
Hey @Malvoz,

We have just encountered a similar issue as @MartijnHarmenzon .

We obviously get the accessibility requirements (and it's actually super nice with a keyboard), but it's causing visual boxes on Google Chrome (not Firefox) by using our mouse and no keyboards, which make things look buggy (we get the "box outline" on a click on a polygon without touching the keyboard from my point of view this is UX/UI bug).
I'll try to build a reproducible example in the coming days.

image

**Comment by Malvoz** [comment]
@FloChehab Please create a new issue for discussion.

**Pull request merged**